src/towncrier/_settings/load.py

```diff
@@ -178,8 +198,13 @@ def parse_toml(base_path: str, config: Mapping[str, Any]) -> Config:
         start_string=config.get("start_string", _start_string),
         title_format=config.get("title_format", _title_format),
         issue_format=config.get("issue_format"),
-        underlines=config.get("underlines", _underlines),
+        underlines=config.get("underlines", _md_underlines if is_md else _underlines),
```

Comment:
I find it easier to read.
```suggestion
        underlines=underlines,
```